app/Listeners/GenericModelCreate.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Listeners;
+
+use App\GenericModel;
+use Illuminate\Support\Facades\App;
+
+class GenericModelCreate
+{
+    /**
+     * Handle the event
+     * @param \App\Events\GenericModelCreate $event
+     */
+    public function handle(\App\Events\GenericModelCreate $event)
+    {
+        $app = App::getFacadeRoot();
```

Comment:
Almost entire logic here is same as in on update listener, extract that to PHP `trait` and use it here